gapis/resolve/command_tree.go

```diff
@@ -512,6 +514,32 @@ func addFrameGroups(ctx context.Context, events *service.Events, p *path.Command
 	}
 }
 
+// setSubCmdNodesName set reasonable names for the intermediate nodes in a
+// group of subcommands. Typically subcommands are group by different pSubmits
+// batches first, and different Command Buffers next.
+func setSubCmdNodesName(
+	ctx context.Context,
+	r *api.SubCmdRoot,
+	refMap map[api.CmdID][]sync.SubcommandReference,
+	commands []api.Cmd) {
+
+	id := api.CmdID(r.Id[0])
+	refs := refMap[id]
+	r.TraverseSubCmdRoot(func(item *api.SubCmdRoot) {
+		switch len(item.Id) {
```

Comment:
This seems a fairly hacky way of distinguishing the different types of nodes. Can we test for the actual conditions we want, or is that data gone by this point?

Reply:
This is not the right place to do this.
We should try and isolate the Vulkan-specific knowledge to the Vulkan package.

Reply:
You can see how we handle debug-marker regions as an example.

Reply:
Thanks for suggestions! Will learn how debug-marker regions do this for a while and may raise some questions during the process. Before finding a better solution will mark this PR as WIP for now.